lib/core/test/unit/Cardano/Wallet/Checkpoints/PolicySpec.hs

```diff
@@ -0,0 +1,134 @@
+{-# LANGUAGE DataKinds #-}
+module Cardano.Wallet.Checkpoints.PolicySpec
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Checkpoints.Policy
+    ( BlockHeight, CheckpointPolicy, nextCheckpoint, toListAtTip )
+import Test.Hspec
+    ( Spec, describe, it )
+import Test.QuickCheck
+    ( Arbitrary (..)
+    , Gen
+    , NonNegative (..)
+    , Property
+    , choose
+    , elements
+    , forAll
+    , frequency
+    , oneof
+    , property
+    , (===)
+    )
+
+import qualified Cardano.Wallet.Checkpoints.Policy as CP
+import qualified Data.Set as Set
+
+spec :: Spec
+spec = do
+    describe "general laws, tested on trailingArithmetic" $
```

Comment:
what about law: we will not have checkpoints beyond epochStability?

Reply:
I have added a property "sparseArithmetic checkpoints after genesis are close to tip". 🤓